automation/test.sh

```diff
@@ -29,15 +29,15 @@ readonly BAZEL_CACHE="${BAZEL_CACHE:-http://bazel-cache.kubevirt-prow.svc.cluste
 
 # Skip if it's docs changes only
 # Only if we are in CI, and this is a non-batch change
-if [[ ${CI} == "true" && -n "$PULL_BASE_SHA" && -n "$PULL_PULL_SHA" ]]; then
+if [[ ${CI} == "true" && -n "$GITHUB_TOKEN" && -n "$PULL_NUMBER" ]]; then
     SKIP_PATTERN="^(docs/)|(OWNERS|OWNERS_ALIASES|.*\.(md|txt))$"
-    CI_GIT_ALL_CHANGES=$(git diff --name-only $PULL_BASE_SHA $PULL_PULL_SHA)
+    CI_GIT_ALL_CHANGES=$(gh pr diff --name-only "https://github.com/kubevirt/kubevirt/pull/$PULL_NUMBER")
```

Comment:
gh has been added to the bootstrap image so it should be available after the next bump of the image.

I've added a check only check the skip pattern if the `gh pr diff` command has run successfully.

Reply:
Thanks but doesnt the -e will fail the script ?

```
$ cat ./try.sh 
#!/bin/bash -e
X=$(false)
echo done
```
won't print done

Reply:
It will fail fast which should be ok. 

```
+ CI=true
+ GITHUB_TOKEN=./unprivileged-token
+ PULL_NUMBER=100000
+ [[ true == \t\r\u\e ]]
+ [[ -n ./unprivileged-token ]]
+ [[ -n 100000 ]]
+ SKIP_PATTERN='^(docs/)|(OWNERS|OWNERS_ALIASES|.*\.(md|txt))$'
+ gh auth login --with-token
++ gh pr diff --name-only https://github.com/kubevirt/kubevirt/pull/100000
could not find pull request diff: HTTP 404: Not Found (https://api.github.com/repos/kubevirt/kubevirt/pulls/100000)
+ CI_GIT_ALL_CHANGES=
$ echo $?
1
```

Reply:
if so we don't even need the condition afterwards, right ?

Reply:
lgtm

Reply:
I just hope we wont get rate limited due to bursts / requests per minute
note that we will also be blocked if API events have down time (assuming github events do continue to reach tide and prow), in either case of those we will be able to change the logic to test instead fail fast in case of an error.